public/app/plugins/datasource/cloudwatch/__mocks__/sqlUtils.ts

```diff
@@ -13,13 +13,13 @@ import {
 export function createArray(
   expressions: QueryEditorExpression[],
   type: QueryEditorExpressionType.And | QueryEditorExpressionType.Or = QueryEditorExpressionType.And
-): QueryEditorArrayExpression {
+) {
   const array = {
     type,
     expressions,
   };
 
-  return array;
+  return array as QueryEditorArrayExpression;
```

Comment:
just a question (non-blocking)

it looks like this type cast is related to the issue you mentioned about nested recursive types? i.e. since `#QueryEditorExpression: #QueryEditorArrayExpression ...` is currently not possible to define recursively in `public/app/plugins/datasource/cloudwatch/dataquery.cue` the `QueryEditorExpression[]` type of the `expressions` argument ends up not being able to satisfy the generated type for `QueryEditorArrayExpression` so we need to type cast it here as a temporary workaround. is my understanding correct?

Reply:
Yes, correct, this should be also fixed when the nested recursive types bug is fixed.